compiler/rustc_middle/src/mir/mod.rs

```diff
@@ -1762,8 +1843,98 @@ pub struct CopyNonOverlapping<'tcx> {
 ///////////////////////////////////////////////////////////////////////////
 // Places
 
-/// A path to a value; something that can be evaluated without
-/// changing or disturbing program state.
+/// Places roughly correspond to a "location in memory." Places in MIR are the same mathematical
+/// object as places in Rust. This of course means that what exactly they are is undecided and part
+/// of the Rust memory model. However, they will likely contain at least the following three pieces
+/// of information in some form:
+///
+///  1. The part of memory that is referred to (see discussion below for details).
+///  2. The type of the place and an optional variant index. See [`PlaceTy`][tcx::PlaceTy]
+///  3. The provenance with which the place is being accessed.
```

Comment:
Having 2 between 1 and 3 seems odd. 1 and 3 together make a pointer (like any pointer *value* in the program) and that is probably worth pointing out.

Reply:
Oh also, places can contain metadata if they are unsized (like pointers).